src/components/campaignguide/Narrator.tsx

```diff
@@ -0,0 +1,550 @@
+          <View
+            style={{
+              display: "flex",
+              flexDirection: "row",
+              height: 64,
+              width: "100%",
+              alignItems: "center",
+              padding: 16,
+              backgroundColor: isCurrentTrack ? "grey" : "transparent",
+            }}
+          >
+            <ArkworkView track={track} state={null} />
+            <TitleView style={{ flex: 1 }} track={track} />
+          </View>
+        </>
+      </TouchableHighlight>
+    );
+  }
+}
+
+interface PlaylistProps {
+  style?: ViewStyle;
+  queue: TrackPlayer.Track[];
+}
+
+interface PlaylistState {
+  currentTrackId: string | null;
+}
+
+class PlaylistView extends React.Component<PlaylistProps, PlaylistState> {
+  onPlaybackTrackChange?: TrackPlayer.EmitterSubscription;
+
+  constructor(props: PlaylistProps) {
+    super(props);
+
+    this.state = {
+      currentTrackId: null,
+    };
+  }
+
+  componentDidMount() {
+    this.onPlaybackTrackChange = TrackPlayer.addEventListener(
+      "playback-track-changed",
+      (data) => {
+        this.setState({
+          currentTrackId: data.nextTrack,
+        });
+      }
+    );
+    TrackPlayer.getCurrentTrack().then((currentTrackId) =>
+      this.setState({ currentTrackId })
+    );
+  }
+
+  componentWillUnmount() {
+    this.onPlaybackTrackChange?.remove();
+  }
+
+  render() {
+    const { style, queue } = this.props;
+    const { currentTrackId } = this.state;
+
+    return (
+      <View style={style}>
+        {queue.map((track) => (
+          <TrackView
+            track={track}
+            isCurrentTrack={currentTrackId === track.id}
+          />
+        ))}
+        <Divider />
+      </View>
+    );
+  }
+}
+
+interface NarratorContainerProps {
+  hasDissonantVoices: boolean;
+  children: JSX.Element,
+}
+
+class NarratorContainerView extends React.Component<
+  NarratorContainerProps,
+  { queue: TrackPlayer.Track[] }
+> {
+  static contextType = StyleContext;
+  context!: StyleContextType;
+  queueHandle: NodeJS.Timeout | null = null;
+
+  constructor(props: NarratorContainerProps) {
+    super(props);
+
+    this.state = {
+      queue: [],
+    };
+  }
+
+  componentDidMount() {
+    this.updateQueue();
+  }
+
+  componentWillUnmount() {
+    if (this.queueHandle !== null) {
+      clearTimeout(this.queueHandle);
+    }
+  }
+
+  updateQueue() {
+    this.queueHandle = setTimeout(async () => {
+      const queue = await TrackPlayer.getQueue();
+      if (!_.isEqual(queue, this.state.queue)) {
+        this.setState(() => ({ queue }), this.updateQueue);
+      }
+    }, 100);
+  }
+
+  render() {
+    const { hasDissonantVoices, children } = this.props;
+    const { queue } = this.state;
+    if (queue.length === 0 || !hasDissonantVoices) return children;
+
+    return (
+      <View style={{height: '100%'}}>
+        {children}
+        <PlayerView />
+        {Platform.OS === 'ios' && <View style={{height: 83}}></View>}
```

Comment:
For some reason on iOS i had to include this otherwise the audio player would be pushed off the screen. Might be something to do with the navigation lib